src/workerd/io/worker.c++

```diff
@@ -1142,6 +1142,59 @@ Worker::Isolate::Isolate(kj::Own<Api> apiParam,
         return v8::MaybeLocal<v8::Promise>();
       }
     });
+
+    // The PRomiseCrossContextResolveCallback is used to ensure that promise reactions
+    // are only scheduled on the microtask queue from the appropriate IoContext for the
+    // promise.
+    lock->v8Isolate->SetPromiseCrossContextResolveCallback(
+        [](v8::Isolate* isolate, v8::Local<v8::Value> tag, v8::Local<v8::Data> reactions,
+            v8::Local<v8::Value> argument,
+            std::function<void(v8::Isolate * isolate, v8::Local<v8::Data> reactions,
+                v8::Local<v8::Value> argument)> callback) {
+      auto& js = jsg::Lock::from(isolate);
+
+      try {
+        auto& ref = jsg::unwrapOpaqueRef<kj::Own<IoContext::WeakRef>>(isolate, tag);
```

Comment:
Ugh I remembered that this doesn't work.

Recapping what we already discussed in person:
* We might be in a different thread.
* The WeakRef is a thread-specific object, we cannot touch it from here. (In fact, storing an `IoContext::WeakRef` in a GC'd object at all can lead to problems.)
* We can't call `context->run()` directly from here either, because we're on the wrong thread.
* Instead, we should extend the `DeleteQueue`, since it is already designed for this sort of cross-thread signalling.
* We could append an item to the `DeleteQueue` whose destructor invokes the code we want. That gets us half-way there.
* But we also have to cause the destination IoContext to wake up promptly to process the queue.
* To that end, each IoContext should create a `kj::newPromiseAndCrossThreadResolver()` when first constructed.
* Put the fulfiller end into the `DeleteQueue`, so that it can be called by the `PromiseCrossContextResolveCallback` from another thread.
* Wait on the promise end. When it resolves, call `run()` to force the `DeleteQueue` to be processed. Then inside the run callback (while the isolate is still locked), replace the promise and fulfiller with a new pair, so that it can be signaled again.